packages/classes/test/index.spec.ts

```diff
@@ -0,0 +1,127 @@
+    it('Should be able to set a new configuration', () => {
+      const newConfig = {
+        baseUrl: 'http://localhost:8080',
+        site: 'editorial',
+        searchId: 'test_search_id',
+        endpoints: {
+          GET_ITEM_URL: '/new_api/content_store/item.json',
+          GET_DESCRIPTOR: '/new_api/content_store/descriptor.json',
+          GET_CHILDREN: '/new_api/content_store/children.json',
+          GET_TREE: '/new_api/content_store/tree.json',
+          GET_NAV_TREE: '/new_api/navigation/tree.json',
+          GET_BREADCRUMB: '/new_api/navigation/breadcrumb.json',
+          TRANSFORM_URL: '/new_api/url/transform.json',
+          SEARCH: '/new_api/search/search.json'
+        }
+      };
+      crafterConf.configure(newConfig);
+      const updatedConfig = crafterConf.getConfig();
+      expect(updatedConfig.site).to.equal(newConfig.site);
+      expect(updatedConfig.baseUrl).to.equal(newConfig.baseUrl);
+      expect(updatedConfig.endpoints.GET_ITEM_URL).to.equal(newConfig.endpoints.GET_ITEM_URL);
+      expect(updatedConfig.endpoints.GET_DESCRIPTOR).to.not.equal('/api/1/content_store/descriptor.json');
+    });
+    // Tests that the mix method returns a mix of the current configuration (set at the previous tes) and the new values
+    // set at this test.
+    it('Should return a mix of the current config and the provided config', () => {
+      const mixConfig = crafterConf.mix({
+        baseUrl: 'http://localhost:3000',
+        site: 'test_site'
+      });
+      expect(mixConfig.site).to.equal('test_site');
+      expect(mixConfig.site).to.not.equal('editorial');
+      expect(mixConfig.baseUrl).to.equal('http://localhost:3000');
+      expect(mixConfig.endpoints.GET_DESCRIPTOR).to.equal('/new_api/content_store/descriptor.json');
+      expect(mixConfig.endpoints.GET_DESCRIPTOR).to.not.equal('/api/1/content_store/descriptor.json');
+    });
+  });
+
+  describe('SDKService', () => {
+    describe('httpGet', () => {
+      // Test the httpGet method with a mocked GET request.
+      it('Should return a response from the GET request', (done) => {
+        nock('http://localhost:8080').get('/api/1/test/getItem').query({ id: 1 }).reply(200, { id: 1, name: 'test' });
+
+        httpGet('http://localhost:8080/api/1/test/getItem?id=1').subscribe((response) => {
+          expect(response).to.not.be.null;
+          expect(response).to.be.an('object');
+          expect(response).to.deep.equal({ id: 1, name: 'test' });
+          done();
+        });
+      });
+    });
+
+    describe('httpPost', () => {
+      // Test the httpPost method with a mocked POST request.
+      it('Should return a response from the POST request', (done) => {
+        nock('http://localhost:8080').post('/api/1/test/addItem').reply(200, { result: 'success', id: 1 });
+
+        httpPost('http://localhost:8080/api/1/test/addItem', { id: 1, name: 'test' }).subscribe((response) => {
```

Comment:
That feels to me more that it would be in API testing, since everything is set up with nock (to mock responses) just by setting it to handle a post, if the request wasn't a post, it would fail beacuse there are no handlers for something other than a POST. That's like an 'implicit' test going on there for both POST and GET.

Reply:
What about the headers?

Reply:
?